internal/validators/base.go

```diff
@@ -0,0 +1,53 @@
+package validators
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"path/filepath"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	apimachyaml "k8s.io/apimachinery/pkg/util/yaml"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Validator interface {
+	Validate(fs.FS) error
+}
```

Comment:
This might be a premature interface definition. I'd expect to see this crop up only when there's a use case around passing different validator implementations around to the same provisioner.